crates/ruma-events/src/room/name.rs

```diff
@@ -18,94 +16,40 @@ pub type NameEvent = StateEvent<NameEventContent>;
 pub struct NameEventContent {
     /// The name of the room.
     #[serde(default, deserialize_with = "ruma_serde::empty_string_as_none")]
-    pub name: Option<RoomName>,
+    pub name: Option<RoomNameBox>,
 }
 
 impl NameEventContent {
     /// Create a new `NameEventContent` with the given name.
-    pub fn new(name: Option<RoomName>) -> Self {
+    pub fn new(name: Option<RoomNameBox>) -> Self {
         Self { name }
     }
 
     /// The name of the room, if any.
     #[deprecated = "You can access the name field directly."]
-    pub fn name(&self) -> Option<&RoomName> {
+    pub fn name(&self) -> Option<&RoomNameBox> {
```

Comment:
```suggestion
    pub fn name(&self) -> Option<&RoomName> {
```

should work with `as_deref` instead of `as_ref` I think.

Reply:
It works, thanks!